Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -0,0 +1,52 @@
+#include "WarpX.H"
+#include "RhoFunctor.H"
+#include "Utils/CoarsenIO.H"
+
+RhoFunctor::RhoFunctor ( const amrex::MultiFab* const mf_src, const int lev,
+                         const amrex::IntVect crse_ratio, bool convertRZmodes2cartesian,
+                         const int ncomp )
+    : ComputeDiagFunctor(ncomp, crse_ratio), m_lev(lev),
+      m_convertRZmodes2cartesian(convertRZmodes2cartesian)
+{
+    // mf_src should not be used, let's make sure it is null
+    AMREX_ALWAYS_ASSERT(mf_src == nullptr);
```

Comment:
You can remove the mf_src argument since it isn't used for anything. This will be cleaner.